luarules/gadgets/include/DirectionsUtil.lua

```diff
@@ -19,86 +19,93 @@ DirectionsUtil.Directions = {}
+
+		repeat                  -- The method begins by sampling & rejecting points.
+			m1 = 2 * math.random() - 1 -- The result can be transformed into radial coords.
+			m2 = 2 * math.random() - 1
+			m3 = m1 * m1 + m2 * m2
+		until (m3 < 1)
+
+		m4 = (1 - m3) ^ 0.5
+		vecs[i    ] = 2 * m1 * m4 -- x
+		vecs[i + 1] = 2 * m2 * m4 -- y
+		vecs[i + 2] = 1 -  2 * m3  -- z
+	end
+
+	return vecs
 end
 
 --------------------------------------------------------------------------------------------------------------
 -- Combined distribution arrays ------------------------------------------------------------------------------
 
---- Fetches a premade direction set from DistributedDirections; otherwise, generates random directions.
--- @tparam[opt] number n count of (unrolled) float3 directions to retrieve
--- @treturn[1] {float,...} array with length equal to 3n
--- @treturn[1] boolean true if the result is random
--- @treturn[2] nil
--- @see DirectionsUtil.GetRandomDirections
+---Fetches a premade direction set from DistributedDirections; otherwise, generates random directions.
+---@param n number count of vectors to retrieve
+---@return table vector3s
+---@return boolean randomized
 DirectionsUtil.GetDirections = function(n)
-    if not n or n < 1 then return end
-    local distributed = DirectionsUtil.Directions[n]
-    if distributed then
-        return distributed, false
-    end
-    if n <= DIRECTION_SET_SIZE_MAX then
-        for ii = 1, 3 * (n - 1) + 1, 3 do
-            return getRandomDirections(n), true
-        end
-    end
+	if not n or n < 1 then return end
+
+	local distributed = DirectionsUtil.Directions[n]
+
+	if distributed then
+		return distributed, false
+	end
+
+	if n <= DIRECTION_SET_SIZE_MAX then
+		for i = 1, 3 * (n - 1) + 1, 3 do
+			return DirectionsUtil.GetRandomDirections(n), true
+		end
+	end
 end
 
---- Pad the size of an insufficient direction set by adding randomly generated directions.
--- @tparam number n maximum solution size to add to Directions
--- @treturn boolean whether provisioning succeeded
+---Pad the size of an insufficient direction set by adding randomly generated directions.
+---@param n number max solution size to add to Directions
+---@return boolean success
 DirectionsUtil.ProvisionDirections = function(n)
-    if n < 2 or n > DIRECTION_SET_SIZE_MAX then return false end
-    local directions = DirectionsUtil.Directions
-    if not directions then return false end
-    for ii = #directions + 1, n do
-        directions[ii] = DirectionsUtil.GetRandomDirections(ii)
-    end
-    return true
+	if n < 2 or n > DIRECTION_SET_SIZE_MAX then return false end
+
+	local directions = DirectionsUtil.Directions
+
+	if not directions then return false end
+
+	for i = #directions + 1, n do
+		directions[i] = DirectionsUtil.GetRandomDirections(i)
+	end
+
+	return true
 end
 
 --------------------------------------------------------------------------------------------------------------
-
```

Comment:
Fixes to whitespace and fake number precision could be in a separate PR.

Reply:
I could do the precision in this PR, it's a general overhaul and cleanup

Reply:
Looking at it further, some numbers have exponents below -20, these might as well be 0

Reply:
Since these are static, computed reference data from a third party, will just leave as-is.
http://neilsloane.com/packings/dim3/